Makefile

```diff
@@ -75,6 +75,20 @@ UFTRACE_LDFLAGS = $(COMMON_LDFLAGS) $(LDFLAGS_$@) $(LDFLAGS_uftrace)
 DEMANGLER_LDFLAGS = $(COMMON_LDFLAGS) $(LDFLAGS_$@) $(LDFLAGS_demangler)
 LIB_LDFLAGS = $(COMMON_LDFLAGS) $(LDFLAGS_$@) $(LDFLAGS_lib) -Wl,--no-undefined
 
+ifeq ($(DEBUG), 1)
+  UFTRACE_CFLAGS += -O0 -g
+  DEMANGLER_CFLAGS += -O0 -g
+  TRACEEVENT_CFLAGS += -O0 -g
+  LIB_CFLAGS += -O0 -g
```

Comment:
I wonder gcc overwrites option value if same option is used more that once.  For example, `COMMON_CFLAGS` has `-O2` but this change will overwrite it to `-O0` right?  Also it could use `COMMON_CFLAGS` directly to set those option (or maybe `CFLAGS_DEBUG` can be used conditionally).